### ISSUE TITLE
cannot change / update the key

### ISSUE THREAD

**Issue by callmeblessed** [aspnet/EntityFramework.Docs#1000]
I followed this but I cannot update the License Plate because it is a key.
How to make it unique but we still able to correct the License Plate later ?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 19fb72e5-5309-7b90-a95d-514a2c62c50c
* Version Independent ID: 1d7d65a5-f8cb-cd45-3dff-34a87da0df2d
* Content: [Alternate Keys (Unique Constraints) - EF Core](https://docs.microsoft.com/en-us/ef/core/modeling/relational/unique-constraints)
* Content Source: [entity-framework/core/modeling/relational/unique-constraints.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/modeling/relational/unique-constraints.md)
* Product: **entity-framework**
* GitHub Login: @rowanmiller
* Microsoft Alias: **divega**

**Comment by ajcvickers** [comment]
@callmeblessed Try using `HasIndex` instead and making the index unique. Alternate keys have additional semantics in EF Core such as not being able to update the property.

**Comment by VictorioBerra** [comment]
I am encountering this issue as well. I can confirm `HasIndex()` combined with  `IsUnique` solved my issue. IMO the docs are misleading. When you google "Unique Constraints" you see that doc first thing. It sounds like you get some unique protection on a column in the database but you dont realize EF wont be able to update it.

**Comment by douglasg14b** [comment]
Agreed with @VictorioBerra THe docs are pretty misleading on this one.

**Comment by davisnw** [comment]
Also, there appears to be no way in EF Core to add a unique *constraint* without having it also be an alternate key.  Sometimes we don't want the extra "alternate key" enforcement that EF adds to the application logic (namely, not being able to change any part of the alternate key), but the dba prefers to have it listed as a unique constraint instead of an index (less likely to accidentally drop an important unique enforcement when evaluating excessive indexes).

**Comment by AndriySvyryd** [comment]
Related: https://github.com/aspnet/EntityFramework.Docs/issues/1561
